### PR TITLE
Udeng 912 add all desktop locales

### PR DESCRIPTION
**Pull request by sergio-costas**
These changes add the missing locales, and also a cut-down version of gnome-language-selector that doesn't need APT to work (although it still requires some touches). This allows to set the locale in ubuntu core desktop. It also requires https://github.com/canonical/ubuntu-desktop-session-snap/pull/21 to work.

### CODE REVIEW

**Comment by kenvandine** [comment]
@sergio-costas any particular reason for using python for the added hook?  Seems this could be done in shell pretty easily to keep this consistent with the other hooks.

**Comment by sergio-costas** [comment]
I'm not so sure... I mean: I can do a for... loop that checks all the .mo files and compares one by one with a list of accepted .mo files; the problem is how to detect, in bash, that a file doesn't match ANY of the files in the list.

Also, I did it in python for clarity... in my opinion, shell scripts are fine for simple tasks, but as soon as what they have to do gets slightly complex, I think that it's better to switch to a clearer script language. A lot of the structures and constructions in shell scripting are really obscure and non-intuitive.

But it's my opinion, of course.

File: hooks/002-add-locales.chroot [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ORIG_BASE_DIR="/usr/share/locale-langpack"
+DEST_BASE_DIR="/usr/share/locale"
+
+apt install -y language-pack-gnome-*
+
+for PATH in "$ORIG_BASE_DIR"/*; do
+    LANG=$(/bin/basename "$PATH")
+    DESTINATION="$DEST_BASE_DIR"/"$LANG"/LC_MESSAGES/
+    /bin/mkdir -p "$DESTINATION"
+    for FILENAME in coreutils.mo ibus10.mo gtk40.mo gtk30.mo gtk20.mo gnome-shell.mo gnome-control-center-2.0.mo gnome-desktop-3.0.mo gnome-initial-setup.mo gnome-menus-3.0.mo language-selector.mo xdg-desktop-portal-gnome.mo xdg-desktop-portal.mo ibus-chewing.mo; do
```
Comment (kenvandine) [inline hooks/002-add-locales.chroot:16]: We should probably ensure we have translations for gdm3, gnome-terminal, and nautilus

**Review by kenvandine — CHANGES_REQUESTED** [review]

**Comment by kenvandine** [comment]
Looks like we also need to include the gkbd-capplet which provides the "Show layout" option in the indicator.  Also, any reason to not remove language-selector-gnome?  Or at least the desktop file it provides?

File: hooks/002-add-locales.chroot (hunk 1 above)
Comment (sergio-costas) [reply hooks/002-add-locales.chroot]: Done. I presumed that those were in their own snap.

**Comment by sergio-costas** [comment]
I'll remove the language-selector .desktop file. About the gkbd-capplet, I prefer to do a different PR.

**Comment by sergio-costas** [comment]
I tested to manually change the global locale and the global keyboard files, and worked: GDM is shown in the specified language and the keyboard worked.

**Comment by kenvandine** [comment]
Are you going to add gkbd-capplet with a separate PR?

**Comment by sergio-costas** [comment]
> Are you going to add gkbd-capplet with a separate PR?

Yes. I think that is better to have this merged and working, and then fight to make that applet work :-)

**Review by kenvandine — APPROVED** [review]
This is good now and I've verified it works with the corresponding ubuntu-desktop-session locale-control change

**Comment by kenvandine** [comment]
@jhenstridge can you take a look at this?

File: hooks/006-add-gdm.chroot [hunk 2]
```diff
@@ -138,3 +138,6 @@ ln -s ../writable/gdm3/custom.conf /etc/gdm3/custom.conf
 
 # Disable gnome-shell systemd
 rm -f /usr/lib/systemd/user/org.gnome.Shell*
+
+# Remove gnome-language-selector .desktop file
```
Comment (robert-ancell) [inline hooks/006-add-gdm.chroot:142]: Comment is just stating what the line does, would be better to describe why, e.g. "Stop gnome-language-selector showing in dash"

**Review by robert-ancell — APPROVED** [review]
Looks good!

File: hooks/006-add-gdm.chroot (hunk 2 above)
Comment (sergio-costas) [reply hooks/006-add-gdm.chroot:142]: Yep... I'll prepare a patch.

**Pull request merged**